src/components/Button/Button.tsx

```diff
@@ -47,6 +47,8 @@ export interface ButtonProps extends BaseButton {
   monochrome?: boolean;
   /** Icon to display to the left of the button content */
   icon?: React.ReactElement | IconSource;
+  /** does the icon display on the left or right of the content */
+  iconPlacement: 'before' | 'after';
```

Comment:
Oh really? I was thinking the opposite, that before/after would allow rtl to render more consistently by flipping the order naturally - although I haven't tested this at all yet.

Reply:
Were you able to test this?

Reply:
yup!

it's these ones: https://github.com/Shopify/polaris-react/pull/3892/files#diff-cbc39de89ade23226f19a812c91b1227646fed602b9b6e7a382c64d00fd8ee1eR164

A bit gross, but seemed like the only realistic way to do it without changing/adding to the quilt testing libs

Reply:
Oh yea meant testing rtl, no I'll try to do that right now.

Reply:
So the padding / borders get all messed up when switching to rtl, which seems like it was already the case before my change too. Going to clean it up a bit to see if I can address.

<img width="614" alt="Screen Shot 2021-03-17 at 10 32 26" src="https://user-images.githubusercontent.com/1128934/111484538-3e544700-870c-11eb-9a63-16ac9c538c48.png">

Reply:
So I spent a bit of time playing with different options regarding the rtl/bidi question. I think before and after does make sense as far as naming goes, as it means that the placement property describes where the icon will end up consistent with the reading direction.

That being said it doesn't seem that polaris handles bidi well, or even at all? I searched through the code and basically didn't find any references to logic that handles bidi yet.

There are a lot of spacing/padding/margin issues in the current implementation of many components that make rendering very `ltr` centric, addressing which would be a very large undertaking. @kyledurand what are your thoughts?